assets/src/components/Sidebar/NewThemeSettings.js

```diff
@@ -51,31 +51,37 @@ const useJsonThemes = () => {
   return jsonThemes;
 }
 
+export const applyChangesToDom = (theme, initialVars) => {
+  if (!theme) {
+    return;
+  }
+  Object.entries(theme).forEach(([name, value]) => {
+    // This will only work reliably if no other code is adding new custom properties to the root element after this
+    // component is first rendered. This should be the case in the post editor.
+    document.documentElement.style.setProperty(name, value);
+  });
+
+  if (initialVars.length === 0) {
+    return;
+  }
+
+  const customProps = getAllDefinedProps();
+
+  customProps.forEach(k => {
+    if (!Object.keys(theme).includes(k) && !initialVars.includes(k)) {
```

Comment:
What do you think to change to this?
`if (![...Object.keys(theme), ....initialVars].includes(k))`
In fact, you don't want to have `k` in any of those `Arrays`.

Reply:
For now I just moved existing logic, so I'd rather not touch it, but I'll definitely look into that deeper when we address the sidebar class component. It will also involve changing `NewThemeSettings`..